Source/Tests/IRLoader/Loader.cpp

```diff
@@ -0,0 +1,498 @@
+    case DecodeFailure::DECODE_UNKNOWN_SSA: return "Unknown SSA value";
+    case DecodeFailure::DECODE_INVALID_CONDFLAG: return "Invalid Conditional name";
+  };
+}
+
+}
+
+namespace FEX::IRLoader {
+  std::unordered_map<std::string_view, FEXCore::IR::IROps> NameToOpMap;
+
+  template<>
+  std::pair<DecodeFailure, uint8_t> Loader::DecodeValue(std::string &Arg) {
+    if (Arg.at(0) != '#') return {DecodeFailure::DECODE_INVALIDCHAR, 0};
+
+    uint8_t Result = strtoul(&Arg.at(1), nullptr, 0);
+    if (errno == ERANGE) return {DecodeFailure::DECODE_INVALIDRANGE, 0};
+    return {DecodeFailure::DECODE_OKAY, Result};
+	}
+
+  template<>
+  std::pair<DecodeFailure, uint16_t> Loader::DecodeValue(std::string &Arg) {
+    if (Arg.at(0) != '#') return {DecodeFailure::DECODE_INVALIDCHAR, 0};
+
+    uint16_t Result = strtoul(&Arg.at(1), nullptr, 0);
+    if (errno == ERANGE) return {DecodeFailure::DECODE_INVALIDRANGE, 0};
+    return {DecodeFailure::DECODE_OKAY, Result};
+  }
+
+  template<>
+  std::pair<DecodeFailure, uint32_t> Loader::DecodeValue(std::string &Arg) {
+    if (Arg.at(0) != '#') return {DecodeFailure::DECODE_INVALIDCHAR, 0};
+
+    uint32_t Result = strtoul(&Arg.at(1), nullptr, 0);
+    if (errno == ERANGE) return {DecodeFailure::DECODE_INVALIDRANGE, 0};
+    return {DecodeFailure::DECODE_OKAY, Result};
+  }
+
+  template<>
+  std::pair<DecodeFailure, uint64_t> Loader::DecodeValue(std::string &Arg) {
+    if (Arg.at(0) != '#') return {DecodeFailure::DECODE_INVALIDCHAR, 0};
+
+    uint64_t Result = strtoull(&Arg.at(1), nullptr, 0);
+    if (errno == ERANGE) return {DecodeFailure::DECODE_INVALIDRANGE, 0};
+    return {DecodeFailure::DECODE_OKAY, Result};
+  }
+
+  template<>
+  std::pair<DecodeFailure, FEXCore::IR::RegisterClassType> Loader::DecodeValue(std::string &Arg) {
+    if (Arg == "GPR") {
+      return {DecodeFailure::DECODE_OKAY, FEXCore::IR::GPRClass};
+    }
+    else if (Arg == "FPR") {
+      return {DecodeFailure::DECODE_OKAY, FEXCore::IR::FPRClass};
+    }
+    else if (Arg == "GPRPair") {
+      return {DecodeFailure::DECODE_OKAY, FEXCore::IR::GPRPairClass};
+    }
+    else if (Arg == "Complex") {
+      return {DecodeFailure::DECODE_OKAY, FEXCore::IR::ComplexClass};
+    }
+
+    return {DecodeFailure::DECODE_INVALIDREGISTERCLASS, FEXCore::IR::InvalidClass};
+  }
+
+  template<>
+  std::pair<DecodeFailure, FEXCore::IR::TypeDefinition> Loader::DecodeValue(std::string &Arg) {
+    uint8_t Size{}, Elements{};
+    int NumArgs = sscanf(Arg.c_str(), "i%hhdv%hhd", &Size, &Elements);
+
+    if (NumArgs != 1 && NumArgs != 2) {
+      return {DecodeFailure::DECODE_INVALID, {}};
+    }
+
+    return {DecodeFailure::DECODE_OKAY, FEXCore::IR::TypeDefinition::Create(Size / 8, Elements)};
+  }
+
+  template<>
+  std::pair<DecodeFailure, FEXCore::IR::CondClassType> Loader::DecodeValue(std::string &Arg) {
+    std::array<std::string, 14> CondNames = {
+      "EQ",
```

Comment:
These strings really should be defined somewhere near the CondClassType enum.

Reply:
I'll make a followup PR for this, since currently the CondClassType definition is defined in JSON. Needs a bit of JSON mangling to support it correctly (Or just move it outside of JSON)